File: Python/plan/create_plan.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.models import *
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

plans_controller = plans_controller.PlansController()

request = create_plan_request.CreatePlanRequest()
request.name = "Plano Gold"
request.currency = "BRL"
request.interval = "month"
request.interval_count = 3
request.billing_type = "prepaid"
request.minimum_price = 10000
request.installments = [3]
request.payment_methods = ["credit_card", "boleto"]
request.items = [create_plan_item_request.CreatePlanItemRequest(), create_plan_item_request.CreatePlanItemRequest()]
# Plan Item 1
request.items[0].name = "Musculação"
request.items[0].quantity = 1
request.items[0].pricing_scheme = create_pricing_scheme_request.CreatePricingSchemeRequest()
request.items[0].pricing_scheme.price = 18990
request.items[0].price = 18990
# Plan Item 2
request.items[1].name = "Matrícula"
# Matrícula ira cobrar apenas 1 vez. Após a primeira cobrança, nao será mais cobrado
request.items[1].cycles = 1
request.items[1].quantity = 1
request.items[1].pricing_scheme
request.items[1].price = 18990
request.items[1].pricing_scheme = create_pricing_scheme_request.CreatePricingSchemeRequest()
request.items[1].pricing_scheme.price = 5990

try:
    result = plans_controller.create_plan(request)
    assert result is not None
    print("Plan Id: ", result.id)
except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex
    

File: Java/plan/CreatePlan.java
package examples.plan;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.PlansController;
import me.pagar.api.http.client.*;
import me.pagar.api.models.*;

import java.util.*;

public class CreatePlan {

    public static void main(String[] args) {

        // Configuration parameters and credentials
        String basicAuthUserName = "basicAuthUserName"; // The username to use with basic authentication
        String basicAuthPassword = "basicAuthPassword"; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        PlansController plans_controller = new PlansController();

        CreatePlanRequest request = new CreatePlanRequest();

        request.setName("Plano Gold");
        request.setCurrency("BRL");
        request.setInterval("month");
        request.setIntervalCount(3);
        request.setBillingType("prepaid");
        request.setMinimumPrice(10000);

        int[] numberOfInstallments = { 3 };
        List<Integer> listInstallments = new ArrayList<>(numberOfInstallments.length);
        for (int installments : numberOfInstallments) {
            listInstallments.add(installments);
        }
        request.setInstallments(listInstallments);

        String[] typesOfPaymentMethods = { "credit_card", "boleto" };
        List<String> listPaymentMethods = new ArrayList<>(typesOfPaymentMethods.length);
        listPaymentMethods.addAll(Arrays.asList(typesOfPaymentMethods));
        request.setPaymentMethods(listPaymentMethods);

        CreatePlanItemRequest planItemOne = new CreatePlanItemRequest();
        planItemOne.setName("Musculação");
        planItemOne.setQuantity(1);
        planItemOne.setPricingScheme(new CreatePricingSchemeRequest());
        planItemOne.getPricingScheme().setPrice(18990);

        CreatePlanItemRequest planItemTwo = new CreatePlanItemRequest();
        planItemTwo.setName("Matrícula");
        planItemTwo.setCycles(1);
        planItemTwo.setQuantity(1);
        planItemTwo.setPricingScheme(new CreatePricingSchemeRequest());
        planItemTwo.getPricingScheme().setPrice(18990);

        List<CreatePlanItemRequest> listPlan = new ArrayList<CreatePlanItemRequest>();
        listPlan.add(planItemOne);
        listPlan.add(planItemTwo);
        request.setItems(listPlan);

        plans_controller.createPlanAsync(request, null, new APICallBack<GetPlanResponse>() {
            @Override
            public void onSuccess(HttpContext context, GetPlanResponse response) {

                System.out.println("Plan create !");
                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println("Plan result status: " + response.getStatus());
                System.out.println("Plan id: " + response.getId());
                System.out.println("Type plan: " + response.getBillingType());

            }

            @Override
            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();

            }
        });
    }
}